### ISSUE TITLE
Add support for SRV records into the NS1 (nsone) Provider

### ISSUE THREAD

**Issue by spkane** [hashicorp/terraform#13744]
### Terraform Version
Terraform v0.8.8

### Affected Resource(s)
- nsone_record

### Terraform Configuration Files
Pseudo-code example:

```hcl
resource "nsone_record" "etcd_srv" {
  zone   = "example.com"
  domain = "_etcd-server._tcp.example.com"
  type   = "SRV"

  answers {
    answer = [10,0,2380,"node-1.example.com"],
    answer = [10,0,2380,"node-2.example.com"]
    answer = [10,0,2380,"node-3.example.com"]
    answer = [10,0,2380,"node-4.example.com"]
    answer = [10,0,2380,"node-5.example.com"]
  }
}
```

### Expected Behavior
That it is possible to create SRV records with the provider.

### Actual Behavior
This is un-supported currently.

### Important Factoids
SRV records can be very important when setting up automated environments with components like etcd or elasticsearch. Supporting this would help significantly is being able to ensure this important configuration step can be included in the overall environment code.

**Comment by pashap** [comment]
This is a failure on my part to correctly document the provider, sorry about that. Answers should be space delimited strings, not `List of RDATA fields` that the current provider docs state. 

Working on the documentation now, hope to have a PR up soon


```tf
resource "ns1_record" "etcd_srv" {
  zone   = "example.com"
  domain = "_etcd-server._tcp.example.com"
  type   = "SRV"

  answers {
    answer = "10 0 2380 node-1.example.com"
  }
  answers {
    answer = "10 0 2380 node-2.example.com"
  }
  answers {
    answer = "10 0 2380 node-3.example.com"
  }
  answers {
    answer = "10 0 2380 node-4.example.com"
  }
  answers {
    answer = "10 0 2380 node-5.example.com"
  }
}
```

Reference https://github.com/hashicorp/terraform/issues/12485 <- same issue(poor provider docs)

**Comment by stack72** [comment]
Closed via documentation updates in #13786
